src/backendMiddleware.ts

```diff
@@ -40,4 +66,120 @@ export class BackendMiddleware {
+    throw new BackendError(ErrorCodes.NoKeyProvider)
+  }
+
+  public async prepareIdentityWallet(): Promise<IdentityWallet> {
+    if (this._identityWallet) return this._identityWallet
+
+    const encryptedEntropy = await this.storageLib.get.encryptedSeed()
+    if (!encryptedEntropy) throw new BackendError(ErrorCodes.NoEntropy)
+    const encryptionPass = await this.keyChainLib.getPassword()
+
+    const decryptedSeed = this.encryptionLib.decryptWithPass({
+      cipher: encryptedEntropy,
+      pass: encryptionPass,
+    })
+
+    if (!decryptedSeed) {
+      throw new BackendError(ErrorCodes.DecryptionFailed)
+    }
+
+    // TODO: rework the seed param on lib, currently cleartext seed is being passed around. Bad.
+    this._keyProvider = JolocomLib.KeyProvider.fromSeed(
+      Buffer.from(decryptedSeed, 'hex'),
+      encryptionPass,
+    )
+
+    const userPubKey = this._keyProvider.getPublicKey({
+      derivationPath: KeyTypes.jolocomIdentityKey,
+      encryptionPass,
+    })
+
+    const didDocument = await this.storageLib.get.didDoc(
+      publicKeyToDID(userPubKey),
+    )
+
+    if (didDocument) {
+      const identity = Identity.fromDidDocument({ didDocument })
+
+      // TODO Simplify constructor
+      return (this._identityWallet = new IdentityWallet({
+        identity,
+        vaultedKeyProvider: this._keyProvider,
+        publicKeyMetadata: {
+          derivationPath: KeyTypes.jolocomIdentityKey,
+          keyId: identity.publicKeySection[0].id,
+        },
+        contractsAdapter: this.registry.contractsAdapter,
+        contractsGateway: this.registry.contractsGateway,
+      }))
+    } else {
+      const { jolocomIdentityKey: derivationPath } = JolocomLib.KeyTypes
+      const identityWallet = await this.registry.authenticate(
+        this._keyProvider,
+        {
+          encryptionPass,
+          derivationPath,
+        },
+      )
+
+      await this.storageLib.store.didDoc(identityWallet.didDocument)
+      return (this._identityWallet = identityWallet)
+    }
+  }
+
+  public async createKeyProvider(encodedEntropy: string): Promise<void> {
+    const password = (await generateSecureRandomBytes(32)).toString('base64')
+    // TODO: rework the seed param on lib, currently cleartext seed is being passed around. Bad.
+    this._keyProvider = JolocomLib.KeyProvider.fromSeed(
+      Buffer.from(encodedEntropy, 'hex'),
+      password,
+    )
+  }
+
+  public async fuelKeyWithEther(): Promise<void> {
+    const password = await this.keyChainLib.getPassword()
+    await JolocomLib.util.fuelKeyWithEther(
+      this.keyProvider.getPublicKey({
+        encryptionPass: password,
+        derivationPath: JolocomLib.KeyTypes.ethereumKey,
+      }),
+    )
+  }
+
+  public async createIdentity(): Promise<Identity> {
+    const password = await this.keyChainLib.getPassword()
+    this._identityWallet = await this.registry.create(
+      this.keyProvider,
+      password,
+    )
+
+    const personaData = {
```

Comment:
I think there is no use of this data anymore, we should discuss what we'll do with this table.

Reply:
Actually, this is probably how we should find out about the user's DID as opposed to using  `publicKeyToDID` as you [mentioned](https://github.com/jolocom/smartwallet-app/pull/1424#discussion_r317592859)! What do you think?

Reply:
actually not a bad idea :+1: Maybe the naming could be improved as it does not really fit the original idea of personas as far as it know.